services/credential-server-ui/src/pages/ConnectionPage.tsx

```diff
@@ -12,16 +12,24 @@ import { UUID_REGEX } from "../constants";
 
 const ConnectionPage: React.FC = () => {
   const [contacts, setContacts] = useState<Contact[]>([]);
+  const [keyword, setKeyword] = useState<string>("");
 
   useEffect(() => {
     handleGetContacts();
   }, []);
 
+  useEffect(() => {
+    handleGetContacts();
+  }, [keyword]);
+
   const handleGetContacts = async () => {
-    console.log("get")
-    setContacts(
-      (await axios.get(`${config.endpoint}${config.path.contacts}`)).data.data
-    );
+    const contacts = (await axios.get(`${config.endpoint}${config.path.contacts}`)).data.data;
+    if (keyword) {
```

Comment:
Can we not fetch the entire list always, store it, and then filter the returned item as keyword/connectionsFilter is updated...? Otherwise each time we change the filter we have to call `handleGetContacts`